linode_metadata/metadata_client.py

```diff
@@ -110,6 +112,38 @@ def check_connection(self):
                 "Can't access Metadata service. Please verify that you are inside a Linode."
             ) from e
 
+    def check_token(self):
+        """
+        Check whether the token is valid. Refresh the token if
+        it's expired and managed by this package.
+        """
+        # We should implicitly refresh the token if the user is enrolled in
+        # token management and the token has expired.
+        if self._managed_token and (
+            self._token is None or datetime.now() >= self._managed_token_expiry
+        ):
+            self.refresh_token(
+                expiry_seconds=self._managed_token_expiry_seconds
+            )
+
+        if self._token is None:
+            raise RuntimeError(
+                "No token provided. Please use "
+                "MetadataClient.refresh_token() to create new token."
+            )
+
+    def get_callable_method(self, method: str) -> Optional[Callable[..., Response]]:
```

Comment:
Just to improve the readability a bit, it might make sense to name this function something like `_get_http_method` :+1: